sdk/keyvault/keyvault-admin/src/backupClient.ts

```diff
@@ -235,39 +240,44 @@ export class KeyVaultBackupClient {
    *
    * const blobStorageUri = "<blob-storage-uri>";
    * const sasToken = "<sas-token>";
-   * const folderName = "<folder-name>";
    * const keyName = "<key-name>";
-   * const poller = await client.beginSelectiveRestore(blobStorageUri, sasToken, folderName, keyName);
+   * const poller = await client.beginSelectiveRestore(keyName, blobStorageUri, sasToken);
    *
    * // Serializing the poller
    * //
    * //   const serialized = poller.toString();
    * //
    * // A new poller can be created with:
    * //
-   * //   await client.beginSelectiveRestore(blobStorageUri, sasToken, folderName, keyName, { resumeFrom: serialized });
+   * //   await client.beginSelectiveRestore(keyName, blobStorageUri, sasToken, { resumeFrom: serialized });
    * //
    *
    * // Waiting until it's done
    * await poller.pollUntilDone();
    * ```
    * Creates a new role assignment.
+   * @param keyName - The name of the key that wants to be restored.
    * @param folderUri - The URL of the blob storage resource, with the folder name of the blob where the previous successful full backup was stored.
    * @param sasToken - The SAS token.
-   * @param folderName - The Folder name of the blob where the previous successful full backup was stored. The URL segment after the container name.
-   * @param keyName - The name of the key that wants to be restored.
    * @param options - The optional parameters.
    */
   public async beginSelectiveRestore(
+    keyName: string,
     folderUri: string,
     sasToken: string,
-    folderName: string,
-    keyName: string,
     options: KeyVaultBeginBackupOptions = {}
   ): Promise<PollerLike<KeyVaultSelectiveRestoreOperationState, KeyVaultRestoreResult>> {
+    const uriParts = folderUri.split("/");
+    const folderName = uriParts.pop();
+    const storageUri = uriParts.join("/");
+
+    if (!folderName) {
+      throw new Error("The provided folder URI is missing the folder name.");
+    }
```

Comment:
Nit: might be worth refactoring this duplicated code into a separate function. It's not like it's much code, but good business logic to make sure keeps in sync.